uiowahjmjohnsonselt2023/selt2023-final-project-team-001
# Seller verification

## New functionality
This branch allows admins to review and approve every seller's request to open a storefront. This feature brings more security and reliability to the Shoppr community, and incentivizes sellers to establish a positive relationship with their buyers.

## Examples
After a user registers as a seller, they will see the option to request a storefront in the user dropdown menu:
<img width="1203" alt="req_dropdown" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/aacc7f92-4118-4511-aa5b-75958ec127b2">

They will be taken to a page where they can read about the review process for sellers creating a storefront (in this example, the user's name is "No Storefront" - the dropdown label still displays user name):
<img width="1176" alt="request_page" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/85aec8c7-2051-4688-883f-a8dd0a245105">

If they check the boxes and hit "Request", then an admin will receive an email with a link to review all pending storefront requests:
<img width="900" alt="admin_receive_email" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/17091002-5576-4367-8c2e-bc560cddc821">

While the user is waiting for a response, their request status is pending. During this time, they cannot request to open another storefront:
<img width="1163" alt="status_pending" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/8c49c5f5-878c-4318-b5fe-b2c097dd42a4">

When an admin follows the email link, they will be taken to a page where they can view the profiles of sellers who have requested a storefront, and accept or reject the requests:
<img width="1167" alt="reqs_page" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/cea048ed-8749-4177-bd83-ded9b470a302">

After approving a request, they are taken back to the requests page to continue reviewing requests, if any are left.

When a user's request is approved, they receive an email to let them know:
<img width="889" alt="approved_email" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/4dc128a1-15c6-4b2b-b93a-a9d6bb2a71f8">

and the next time they visit the site, they will have the "create storefront" option in their user dropdown menu (again, this is user "No Storefront"):
<img width="1208" alt="approved_dropdown" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/24beee12-0080-483a-98ce-d9c381ed4c94">
 ## Sad paths
A user who's rating is less than 3 stars or who has fewer than five reviews will not be able to request a storefront at all:
<img width="1155" alt="rating_too_low" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/829f3375-15e3-4edf-a00b-57928d604f97">

<img width="1184" alt="not_enough_reviews" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/63525fff-341c-4326-b952-b587cac5de10">

A user who is not logged in will not be able to view storefront requests:
<img width="1167" alt="admin_link_while_logged_out" src="https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/assets/80122119/2869fb2b-cdc8-4950-91dd-cc98231c253c">

And similarly, a user who is not an admin will not be able to view storefront requests. 

## To test
Create two users who are registered as sellers using an email you have access to - you can use the same email address if you run `bin/rake db:reset` and then `bin/rake db:seed` in between tests. When logged in as each of these sellers, request a storefront. 
Check the admin email address shopprwebsite@gmail.com to see that two emails have come in about requests. Log in as an admin on the Shoppr website, and then follow the link in the email to see the requests home page. 
Approve one seller and reject the other. Check the email for these users to see the response messages. Log into the app as either of these users and verify that you either can create a storefront, or cannot even request a storefront. 

## Specs
Coverage report generated for RSpec to /Users/jkessler/Desktop/3_FA_23/selt/SELT_REPOS/selt2023-final-project-team-001/coverage. 2240 / 2428 LOC (92.26%) covered.
